test/inspect.test.js

```diff
@@ -82,313 +321,365 @@ test('inspect', function (t) {
-  chdirWorkspaces('pip-app-trusted-host');
-  return pipInstall()
-    .then(function () {
-      return plugin.inspect('.', 'requirements.txt')
-        .then(function (result) {
-          t.ok(result.package.dependencies, 'does not error');
-          t.end();
-        });
-    })
-    .catch(function (error) {
-      t.fail(error);
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pip-app-trusted-host');
+    var venvCreated = ensureVirtualenv('pip-app-trusted-host');
+    activateVirtualenv('pip-app-trusted-host');
+    if (venvCreated) {
+      pipInstall();
+    }
+  })
+  .then(function () {
+    return plugin.inspect('.', 'requirements.txt')
+  })
+  .then(function (result) {
+    t.ok(result.package.dependencies, 'does not error');
+    t.end();
+  })
+  .finally(deactivateVirtualenv);
+});
+
+test('inspect Pipfile', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipfile-pipapp');
+    activateVirtualenv('pip-app');
+  })
+  .then(function () {
+    return plugin.inspect('.', 'Pipfile')
+  })
+  .then(function (result) {
+    var pkg = result.package;
+
+    t.test('package dependencies', function (t) {
+      t.notOk(pkg.dependencies['django'], 'django skipped (editable)');
+
+      t.match(pkg.dependencies['django-select2'], {
+        name: 'django-select2',
+        version: '6.0.1',
+        dependencies: {
+          'django-appconf': {
+            name: 'django-appconf',
+          },
+        },
+      }, 'django-select2 looks ok');
+
+      t.match(pkg.dependencies['python-etcd'], {
+        name: 'python-etcd',
+        version: /^0\.4.*$/,
+      }, 'python-etcd looks ok');
+
+      t.notOk(pkg.dependencies['e1839a8'],
+        'dummy local package skipped (editable)');
+
+      t.ok(pkg.dependencies['jinja2'] !== undefined, 'jinja2 found');
+      t.ok(pkg.dependencies['testtools'] !== undefined, 'testtools found');
+
+      t.end();
     });
+
+    t.end();
+  })
+  .finally(deactivateVirtualenv);
+});
+
+test('inspect Pipfile with pinned versions', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipfile-pipapp-pinned');
+    activateVirtualenv('pip-app');
+  })
+  .then(function () {
+    return plugin.inspect('.', 'Pipfile')
+  })
+  .then(function (result) {
+    var pkg = result.package;
+
+    t.test('package dependencies', function (t) {
+      Object.keys(pipAppExpectedDependencies).forEach(function (depName) {
+        t.match(
+          pkg.dependencies[depName],
+          pipAppExpectedDependencies[depName].data,
+          pipAppExpectedDependencies[depName].msg
+        );
+      });
+
+      t.end();
+    });
+
+    t.end();
+  })
+  .finally(deactivateVirtualenv);
+});
+
+test('inspect pipenv app', function (t) {
+  return Promise.resolve().then(function () {
+    chdirWorkspaces('pipenv-app');
+
+    var venvCreated = ensureVirtualenv('pipenv-app');
+    activateVirtualenv('pipenv-app');
```

Comment:
so if `pipenv` runs inside a virtualenv it doesn't create it's own kind virtualenv?  don't really understand what's going on here

Reply:
Indeed, it only creates its own virtualenv if it isn't already inside of one. I can add a test for such a scenario, too, if you like. We'll have to "deactivate" the existing virtualenv if the tests were run from inside of one.

Reply:
The test I imagine is a one that let's `pipenv` create it's own virtualenv, and then run `inspect` inside it.